File: logistics_v2/checkout_session_state.py
from __future__ import annotations

from enum import Enum


class CheckoutSessionState(str, Enum):
    CHECKOUT_STARTED = "checkout_started"
    PAYMENT_CONFIRMED = "payment_confirmed"
    PACKED = "packed"
    SHIPPED = "shipped"
    DELIVERED = "delivered"


_EMOJI = {
    CheckoutSessionState.CHECKOUT_STARTED: "🛒",
    CheckoutSessionState.PAYMENT_CONFIRMED: "💳",
    CheckoutSessionState.PACKED: "📦",
    CheckoutSessionState.SHIPPED: "🚚",
    CheckoutSessionState.DELIVERED: "✅",
}

_LABEL = {
    CheckoutSessionState.CHECKOUT_STARTED: "Checkout started",
    CheckoutSessionState.PAYMENT_CONFIRMED: "Paid",
    CheckoutSessionState.PACKED: "Packed",
    CheckoutSessionState.SHIPPED: "Shipped",
    CheckoutSessionState.DELIVERED: "Delivered",
}


def emoji(state: CheckoutSessionState) -> str:
    return _EMOJI.get(state, "🛒")


def label(state: CheckoutSessionState) -> str:
    return _LABEL.get(state, state.value)


def display(state: CheckoutSessionState) -> str:
    return f"{emoji(state)} {label(state)}"


def to_tracking_status(state: CheckoutSessionState) -> str | None:
    """Maps session state to legacy OrderModel.tracking_status values."""
    return {
        CheckoutSessionState.PAYMENT_CONFIRMED: "paid",
        CheckoutSessionState.PACKED: "packed",
        CheckoutSessionState.SHIPPED: "shipped",
        CheckoutSessionState.DELIVERED: "delivered",
    }.get(state)

File: Baasket.LogisticsV2/CheckoutSessionState.cs
namespace Baasket.LogisticsV2;

/// <summary>
/// Lifecycle states for checkout → payment → logistics tracking.
/// </summary>
public enum CheckoutSessionState
{
    CheckoutStarted = 0,
    PaymentConfirmed = 1,
    Packed = 2,
    Shipped = 3,
    Delivered = 4,
}

public static class CheckoutSessionStateDisplay
{
    public static string Emoji(CheckoutSessionState state) => state switch
    {
        CheckoutSessionState.PaymentConfirmed => "💳",
        CheckoutSessionState.Packed => "📦",
        CheckoutSessionState.Shipped => "🚚",
        CheckoutSessionState.Delivered => "✅",
        _ => "🛒",
    };

    public static string Label(CheckoutSessionState state) => state switch
    {
        CheckoutSessionState.CheckoutStarted => "Checkout started",
        CheckoutSessionState.PaymentConfirmed => "Paid",
        CheckoutSessionState.Packed => "Packed",
        CheckoutSessionState.Shipped => "Shipped",
        CheckoutSessionState.Delivered => "Delivered",
        _ => state.ToString(),
    };

    public static string Display(CheckoutSessionState state) =>
        $"{Emoji(state)} {Label(state)}";
}
